src/utils/usedAndSelectedResources.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ext } from "../extensionVariables";
+
+function nodeHasIdAndMaybeResource(node: unknown): node is { id: string, resource?: { kind?: string } } {
+    // Make sure node is a not-null object
+    return typeof node === "object" && node !== null &&
+        (
+            // Make sure node has an id property that is a string
+            typeof (node as { id: string }).id === "string"
+        ) &&
+        (
+            // Make sure node.resource is undefined or a not-null object
+            (node as { resource?: unknown }).resource === undefined ||
+            (typeof (node as { resource: unknown }).resource === "object" && (node as { resource: unknown }).resource !== null)
+        ) &&
+        (
+            // Make sure node.resource.kind is undefined or a string
+            (node as { resource: { kind?: string } }).resource?.kind === undefined ||
+            typeof (node as { resource: { kind?: string } }).resource?.kind === "string"
+        );
+}
+
+function getArmIdAndMaybeKindFromTreeNode(node: unknown | undefined): { id: string, kind?: string } | undefined {
+    if (node && nodeHasIdAndMaybeResource(node)) {
+        const resourceIdParts = node.id.split("/");
+        const subscriptionsIdx = resourceIdParts.indexOf("subscriptions");
+        const resourceGroupsIdx = resourceIdParts.indexOf("resourceGroups");
+        const providersIdx = resourceIdParts.indexOf("providers");
+        if (subscriptionsIdx === resourceIdParts.length - 8 && resourceGroupsIdx === resourceIdParts.length - 6 && providersIdx === resourceIdParts.length - 4) {
+            const resourceId = resourceIdParts.slice(-8).join("/");
```

Comment:
We typically keep the beginning slash

```suggestion
            const resourceId = "/" + resourceIdParts.slice(-8).join("/");
```